components/src/core/AppBar/AppBar.tsx

```diff
@@ -223,7 +222,7 @@ const AppBarRender: React.ForwardRefRenderFunction<unknown, AppBarProps> = (prop
                 expandToolbar();
             }
         }
-    }, [variant]);
+    }, [variant, collapseToolbar, expandToolbar, offset, scrollThreshold]);
```

Comment:
Changing these dependencies means that this useEffect will run any time one of the dependencies changes. The description of the useEffect says it's intended to run only when the variant changes...has this change impacted functionality?

Reply:
I didn't see any functionality issues. If I remove everything but variant then functionality changes.


https://github.com/etn-ccis/blui-react-component-library/assets/119693939/3b64ae20-6421-4e27-9cad-8b6971e745ed

Reply:
if I only have these deps [variant, offset, scrollThreshold] then it works.


https://github.com/etn-ccis/blui-react-component-library/assets/119693939/695963d8-c169-47f1-8fa8-aa17a0f460de

Reply:
Are you saying the original implementation (`[variant]` only) doesn't work?

Reply:
yes

Reply:
It works in master branch with the original implementation:
https://brightlayer-ui-components.github.io/react/components/app-bar/examples

Reply:
I've changed what the original comment was about and that was to only have 'variant' in the deps array. I changed it back to whats in master and ran showcase and docs and the appBar functionality (expand, collapse) isn't working. seen in the first screen recording.

<img width="386" alt="image" src="https://github.com/etn-ccis/blui-react-component-library/assets/119693939/524f1140-05cf-4e8a-be05-b4f4ef9319a4">

this does not work. You can scroll but no dice on appBar

Reply:
Right. What I'm asking is if you have reverted the other changes (6 questionable hook dependencies mentioned in my review) to what's in master when you are testing or if you only changed this single hook (and left the other questionable changes in place). Because the master branch implementation clearly works — you can see it working in the deployed docs site. So that means some other pending local changes are likely interfering.

I would revert all 6 of these hooks to their original state and then integrate your changes one hook at a time and verify that the behavior is not adversely affected.